climada/hazard/tc_tracks.py

```diff
@@ -1249,12 +1258,70 @@ def from_netcdf(cls, folder_name):
                                "whole life time.")
                 basin = track.basin
                 del track.attrs['basin']
-                track['basin'] = ("time", np.full(track.time.size, basin))
+                track['basin'] = ("time", np.full(track.time.size, basin, dtype="<U2"))
             data.append(track)
         tr = cls()
         tr.data = data
         return tr
 
+    def write_hdf(self, file_name, complevel=4):
+        """Write TC tracks in HDF5 format.
+
+        Parameters
+        ----------
+        file_name: str or Path
+            Path to a new HDF5 file. If it exists already, the file is overwritten.
+        complevel : int
+            Specifies a compression level (0-9) for the gzip compression of the data.
+            A value of 0 or None disables compression. Default: 4
+        """
+        with h5py.File(file_name, 'w') as store:
+            store.create_dataset('size', (), dtype=int, data=self.size)
+
+            LOGGER.info('Writing %d tracks to %s', self.size, file_name)
+            last_perc = 0
+            for i, tr in enumerate(self.data):
+                perc = 100 * i / self.size
+                if perc - last_perc >= 10:
+                    LOGGER.info("Progress: %d%%", perc)
+                    last_perc = perc
+
+                tr['basin'] = tr['basin'].astype('<U2')
+                tr_bytes = tr.to_netcdf()
+                store.create_dataset(
+                    f'track{i}', (1,), dtype=f'|S{len(tr_bytes)}', data=[tr_bytes],
+                    compression='gzip', compression_opts=complevel)
+
+            if last_perc != 100:
+                LOGGER.info("Progress: 100%")
+
+    @classmethod
+    def from_hdf(cls, file_name):
+        """Create new TCTracks object from HDF5 file
+
+        Parameters
+        ----------
+        file_name : str or Path
+            Path to a file that has been generated with `TCTracks.write_hdf`.
+
+        Returns
+        -------
+        tracks : TCTracks
+            TCTracks with data from the given HDF5 file.
+        """
+        data = []
+        with h5py.File(file_name, 'r') as store:
+            size = store.get('size')[()]
+            LOGGER.info('Reading %d tracks from %s', size, file_name)
+            for i in range(size):
+                tr_ds = store.get(f'track{i}')
+                tr_bytes = tr_ds[0].ljust(int(str(tr_ds.dtype)[2:]), b'\x00')
```

Comment:
I do not understand what is going on in this line exactly, but I trust that it is doing the right operation.

Reply:
By default, numpy strips trailing \x00 in byte strings, so we need to undo that. In my latest commit, I mention this in an inline comment.